### PR TITLE
Confirm account

### PR DESCRIPTION
**Pull request by flomosq**
# Confirm account

### Trello Card #16

- Confirmable config
- Confirm user after creation in tests
- Send email confirmation test
- Sendgrid integration

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -48,3 +48,7 @@ end
 
 # Windows does not include zoneinfo files, so bundle the tzinfo-data gem
 gem 'tzinfo-data', platforms: %i[mingw mswin x64_mingw jruby]
+
```
Comment (santir489) [inline Gemfile]: you should remove empty line between gems

File: spec/factories/users.rb [hunk 2]
```diff
@@ -3,5 +3,7 @@
     email
     password { '12345678' }
     gender { 'female' }
+
+    after(:create, &:confirm)
```
Comment (santir489) [inline spec/factories/users.rb:7]: This is a valid option.

Another option is add an attribute like 
```
confirmed_at      { 3.hours.ago }
```
I think the second adds less complexity than an `after_create`

File: spec/requests/users/create_spec.rb [hunk 3]
```diff
@@ -29,6 +29,12 @@
     }.to change(User, :count).by(1)
   end
 
+  it 'sends email confirmation' do
+    expect {
+      subject
+    }.to change { ActionMailer::Base.deliveries.size }.by(1)
```
Comment (santir489) [inline spec/requests/users/create_spec.rb]: I recomend to use count, in this scenarios.
When using `size` you must be aware to not have a preload value.
`count` goes direct to the DB.

File: Gemfile (hunk 1 above)
Comment (brunvez) [reply Gemfile]: And push them to the top. Production gems should be the first thing you see

File: app/models/user.rb [hunk 4]
```diff
@@ -36,7 +36,8 @@ class User < ApplicationRecord
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
```
Comment (brunvez) [inline app/models/user.rb:37]:
```suggestion
   # :lockable, :timeoutable, :trackable and :omniauthable
```

File: spec/factories/users.rb (hunk 2 above)
Comment (brunvez) [reply spec/factories/users.rb:7]: Agree with setting the attributes directly, less DB calls

File: spec/requests/users/create_spec.rb (hunk 3 above)
Comment (brunvez) [reply spec/requests/users/create_spec.rb]: Yeah tbh we never use `#size` or `#length`

File: config/initializers/smtp.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+ActionMailer::Base.smtp_settings = {
+  domain: ENV['SERVER_HOST'],
+  address: 'smtp.sendgrid.net',
```
Comment (martinjaimem) [inline config/initializers/smtp.rb]: you can move this to a constant or an ENV variable, but not a must 👍

**Review by santir489 — APPROVED** [review]

**Review by brunvez — APPROVED** [review]

**Pull request merged**